app/routes/session.py

```diff
@@ -139,11 +141,14 @@ def login() -> Response:
 
 
 def _set_questionnaire_supplementary_data(
-    questionnaire_store: QuestionnaireStore, metadata: MetadataProxy
+    *,
+    questionnaire_store: QuestionnaireStore,
+    metadata: MetadataProxy,
+    schema: QuestionnaireSchema,
 ) -> None:
     """
     If the survey metadata has an sds dataset id, and it either doesn't match what it stored, or there is no stored supplementary data
-    then fetch it and add it to the store
+    then fetch it, verify any schema supplementary lists are included in the fetched data, and add it to the questionnaire store
     """
     if not (new_sds_dataset_id := metadata["sds_dataset_id"]):
```

Comment:
Just noticed a problem. Since we do the validation after this check, if the schema list changes, but the datasets hadn't, we wouldn't validate again, hence it could be invalid but still pass our checks. Edge and would only happen if they update mid-exercise. i.e Author updates the schema, but the dataset id hasn't changed yet and a new list was added/renamed.

Reply:
ahh yes... won't be able to avoid that with the list validation in the parser - could always revert the last commit so its called in `_set_questionnaire_supplementary_data` and additionally validate in the cases of no data / `existing == new` ?

Reply:
Ya we'll need to cover this just in case.

Reply:
Have made the switch, will now validate every launch regardless of new data or not